diff --git a/dist/cjs/cli/configPlugin.js b/dist/cjs/cli/configPlugin.js
index 3f1af86..a038365 100644
--- a/dist/cjs/cli/configPlugin.js
+++ b/dist/cjs/cli/configPlugin.js
@@ -216,9 +216,12 @@ var __webpack_exports__ = {};
         if (!chain.output.get('uniqueName')) chain.output.uniqueName(mfConfig.name);
         const splitChunkConfig = chain.optimization.splitChunks.entries();
         if (!isServer) (0, utils_namespaceObject.autoDeleteSplitChunkCacheGroups)(mfConfig, splitChunkConfig);
-        if (!isServer && enableSSR && splitChunkConfig && 'object' == typeof splitChunkConfig && splitChunkConfig.cacheGroups) {
+        if (!isServer && enableSSR && splitChunkConfig && 'object' == typeof splitChunkConfig && splitChunkConfig.cacheGroups && splitChunkConfig.chunks !== undefined && splitChunkConfig.chunks !== 'async') {
+            const previousChunks = splitChunkConfig.chunks;
             splitChunkConfig.chunks = 'async';
-            external_logger_js_default().warn('splitChunks.chunks = async is not allowed with stream SSR mode, it will auto changed to "async"');
+            if (void 0 !== previousChunks) {
+                external_logger_js_default().warn(`splitChunks.chunks = ${previousChunks} is not allowed with stream SSR mode, it will auto changed to "async"`);
+            }
         }
         if ((0, external_utils_js_namespaceObject.isDev)() && 'auto' === chain.output.get('publicPath')) {
             var _modernjsConfig_server;
diff --git a/dist/esm-node/cli/configPlugin.mjs b/dist/esm-node/cli/configPlugin.mjs
index c238b85..7350973 100644
--- a/dist/esm-node/cli/configPlugin.mjs
+++ b/dist/esm-node/cli/configPlugin.mjs
@@ -156,9 +156,12 @@ function patchBundlerConfig(options) {
     if (!chain.output.get('uniqueName')) chain.output.uniqueName(mfConfig.name);
     const splitChunkConfig = chain.optimization.splitChunks.entries();
     if (!isServer) autoDeleteSplitChunkCacheGroups(mfConfig, splitChunkConfig);
-    if (!isServer && enableSSR && splitChunkConfig && 'object' == typeof splitChunkConfig && splitChunkConfig.cacheGroups) {
+    if (!isServer && enableSSR && splitChunkConfig && 'object' == typeof splitChunkConfig && splitChunkConfig.cacheGroups && splitChunkConfig.chunks !== undefined && splitChunkConfig.chunks !== 'async') {
+        const previousChunks = splitChunkConfig.chunks;
         splitChunkConfig.chunks = 'async';
-        logger.warn('splitChunks.chunks = async is not allowed with stream SSR mode, it will auto changed to "async"');
+        if (void 0 !== previousChunks) {
+            logger.warn(`splitChunks.chunks = ${previousChunks} is not allowed with stream SSR mode, it will auto changed to "async"`);
+        }
     }
     if (isDev() && 'auto' === chain.output.get('publicPath')) {
         var _modernjsConfig_server;
diff --git a/dist/esm/cli/configPlugin.mjs b/dist/esm/cli/configPlugin.mjs
index c8c2ace..fee11da 100644
--- a/dist/esm/cli/configPlugin.mjs
+++ b/dist/esm/cli/configPlugin.mjs
@@ -185,9 +185,12 @@ function patchBundlerConfig(options) {
     if (!chain.output.get('uniqueName')) chain.output.uniqueName(mfConfig.name);
     var splitChunkConfig = chain.optimization.splitChunks.entries();
     if (!isServer) autoDeleteSplitChunkCacheGroups(mfConfig, splitChunkConfig);
-    if (!isServer && enableSSR && splitChunkConfig && (void 0 === splitChunkConfig ? "undefined" : _type_of__(splitChunkConfig)) === 'object' && splitChunkConfig.cacheGroups) {
+    if (!isServer && enableSSR && splitChunkConfig && (void 0 === splitChunkConfig ? "undefined" : _type_of__(splitChunkConfig)) === 'object' && splitChunkConfig.cacheGroups && void 0 !== splitChunkConfig.chunks && splitChunkConfig.chunks !== 'async') {
+        var previousChunks = splitChunkConfig.chunks;
         splitChunkConfig.chunks = 'async';
-        logger.warn('splitChunks.chunks = async is not allowed with stream SSR mode, it will auto changed to "async"');
+        if (void 0 !== previousChunks) {
+            logger.warn("splitChunks.chunks = ".concat(previousChunks, " is not allowed with stream SSR mode, it will auto changed to \"async\""));
+        }
     }
     if (isDev() && 'auto' === chain.output.get('publicPath')) {
         var _modernjsConfig_server;
