diff --git a/dist/remote/index.d.ts b/dist/remote/index.d.ts
index 3bde403c13ae782e9abfbc579c5e61db40aa130f..3a1f44032c3ed8b289b87beb1dbf30d5db33184e 100644
--- a/dist/remote/index.d.ts
+++ b/dist/remote/index.d.ts
@@ -6,7 +6,7 @@ import { AsyncWaterfallHook } from "../utils/hooks/asyncWaterfallHooks.js";
 import { PluginSystem } from "../utils/hooks/pluginSystem.js";
 import { ModuleFederation } from "../core.js";
 import { CallFrom, Options, Remote, RemoteEntryExports, RemoteInfo, UserOptions } from "../type/config.js";
-import { PreloadAssets, PreloadOptions, PreloadRemoteArgs, PreloadRemoteResult } from "../type/preload.js";
+import { PreloadAssets, PreloadOptions, PreloadRemoteArgs, PreloadRemoteResult, ResourceLoadContext } from "../type/preload.js";
 import { GlobalModuleInfo, ModuleInfo } from "@module-federation/sdk";
 
 //#region src/remote/index.d.ts
