diff --git a/dist/lazy/wrapNoSSR.d.ts b/dist/lazy/wrapNoSSR.d.ts
--- a/dist/lazy/wrapNoSSR.d.ts
+++ b/dist/lazy/wrapNoSSR.d.ts
@@ -6,4 +6,4 @@ export declare function wrapNoSSR<T, E extends keyof T>(createLazyComponentFn:
     key?: React.Key | null;
 } : {
     key?: React.Key | null;
-}) => import("node_modules/@types/react").JSX.Element;
+}) => import("react").JSX.Element;
diff --git a/dist/provider/versions/v18.d.ts b/dist/provider/versions/v18.d.ts
--- a/dist/provider/versions/v18.d.ts
+++ b/dist/provider/versions/v18.d.ts
@@ -12,7 +12,7 @@ export interface Root {
     unmount(): void;
 }
 export declare function createReact18Root(container: Element | DocumentFragment, options?: CreateRootOptions): Root;
-export declare function hydrateReact18Root(container: Element | DocumentFragment, initialChildren: React.ReactNode, options?: CreateRootOptions): import("node_modules/@types/react-dom/client").Root;
+export declare function hydrateReact18Root(container: Element | DocumentFragment, initialChildren: React.ReactNode, options?: CreateRootOptions): import("react-dom/client").Root;
 export declare function createBridgeComponent<T = any>(bridgeInfo: Omit<ProviderFnParams<T>, 'createRoot'>): () => {
     render(info: import("../..").RenderParams): Promise<void>;
     destroy(info: import("../..").DestroyParams): void;
diff --git a/dist/remote/base-component/create.d.ts b/dist/remote/base-component/create.d.ts
--- a/dist/remote/base-component/create.d.ts
+++ b/dist/remote/base-component/create.d.ts
@@ -1,7 +1,7 @@
 import { type LazyRemoteComponentInfo } from '../createHelpers';
 export type { LazyRemoteComponentInfo };
-declare const createLazyRemoteComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("node_modules/@types/react").LazyExoticComponent<import("node_modules/@types/react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("node_modules/@types/react").RefAttributes<HTMLDivElement>>>;
-export declare const createRemoteAppComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("node_modules/@types/react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("node_modules/@types/react").RefAttributes<HTMLDivElement>>;
+declare const createLazyRemoteComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("react").LazyExoticComponent<import("react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("react").RefAttributes<HTMLDivElement>>>;
+export declare const createRemoteAppComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("react").RefAttributes<HTMLDivElement>>;
 /**
  * @deprecated createRemoteComponent is deprecated, please use createRemoteAppComponent instead!
  */
diff --git a/dist/remote/router-component/create.d.ts b/dist/remote/router-component/create.d.ts
--- a/dist/remote/router-component/create.d.ts
+++ b/dist/remote/router-component/create.d.ts
@@ -1,7 +1,7 @@
 import { type LazyRemoteComponentInfo } from '../createHelpers';
 export type { LazyRemoteComponentInfo };
-declare const createLazyRemoteComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("node_modules/@types/react").LazyExoticComponent<import("node_modules/@types/react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("node_modules/@types/react").RefAttributes<HTMLDivElement>>>;
-export declare const createRemoteAppComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("node_modules/@types/react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("node_modules/@types/react").RefAttributes<HTMLDivElement>>;
+declare const createLazyRemoteComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("react").LazyExoticComponent<import("react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("react").RefAttributes<HTMLDivElement>>>;
+export declare const createRemoteAppComponent: <T = Record<string, unknown>, E extends keyof T = keyof T>(info: LazyRemoteComponentInfo<T, E>) => import("react").ForwardRefExoticComponent<Omit<import("../..").RemoteComponentProps<Record<string, unknown>>, "ref"> & import("react").RefAttributes<HTMLDivElement>>;
 /**
  * @deprecated createRemoteComponent is deprecated, please use createRemoteAppComponent instead!
  */
