diff --git a/dist/esm/expose-rpc-B0rqtOKP.mjs b/dist/esm/expose-rpc-B0rqtOKP.mjs
index 1692a3e..1c99630 100644
--- a/dist/esm/expose-rpc-B0rqtOKP.mjs
+++ b/dist/esm/expose-rpc-B0rqtOKP.mjs
@@ -1415,7 +1415,7 @@ const formatCompilerError = (error) => {
 const getDependentFilesWithTsc = (rootFiles, rootDir, resolvedTsConfigPath, compilerOptions, context, typeScriptContext) => {
 	if (!rootFiles.length) return [];
 	const typeScriptPackageInfo = getTypeScriptPackageInfo(typeScriptContext);
-	const listFilesTsConfigPath = writeListFilesTsConfig(rootFiles, resolvedTsConfigPath, context, compilerOptions);
+	const listFilesTsConfigPath = writeListFilesTsConfig(rootFiles, resolvedTsConfigPath, context, { ...compilerOptions, rootDir });
 	try {
 		const dependentFiles = execFileSync(process.execPath, [
 			typeScriptPackageInfo.tscBinPath,
diff --git a/dist/expose-rpc-jaGpsAVL.js b/dist/expose-rpc-jaGpsAVL.js
index 55f07d8..7828615 100644
--- a/dist/expose-rpc-jaGpsAVL.js
+++ b/dist/expose-rpc-jaGpsAVL.js
@@ -1404,7 +1404,7 @@ const formatCompilerError = (error) => {
 const getDependentFilesWithTsc = (rootFiles, rootDir, resolvedTsConfigPath, compilerOptions, context, typeScriptContext) => {
 	if (!rootFiles.length) return [];
 	const typeScriptPackageInfo = getTypeScriptPackageInfo(typeScriptContext);
-	const listFilesTsConfigPath = writeListFilesTsConfig(rootFiles, resolvedTsConfigPath, context, compilerOptions);
+	const listFilesTsConfigPath = writeListFilesTsConfig(rootFiles, resolvedTsConfigPath, context, { ...compilerOptions, rootDir });
 	try {
 		const dependentFiles = (0, child_process.execFileSync)(process.execPath, [
 			typeScriptPackageInfo.tscBinPath,
