diff --git a/dist/cjs/load-matches.cjs b/dist/cjs/load-matches.cjs
index c09976a..c08e017 100644
--- a/dist/cjs/load-matches.cjs
+++ b/dist/cjs/load-matches.cjs
@@ -439,6 +439,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 				try {
 					await runLoader(inner, matchPromises, matchId, index, route);
 					const match = inner.router.getMatch(matchId);
+					if (!match) return;
 					match._nonReactive.loaderPromise?.resolve();
 					match._nonReactive.loadPromise?.resolve();
 					match._nonReactive.loaderPromise = void 0;
@@ -469,6 +470,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 			if (prevMatch.status === "success" && !inner.sync && !prevMatch.preload && shouldReloadInBackground) return prevMatch;
 			await prevMatch._nonReactive.loaderPromise;
 			const match = inner.router.getMatch(matchId);
+			if (!match) return inner.matches[index];
 			const error = match._nonReactive.error || match.error;
 			if (error) handleRedirectAndNotFound(inner, match, error);
 			if (match.status === "pending") await handleLoader(preload, prevMatch, previousRouteMatchId, match, route);
@@ -484,6 +486,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 		}
 	}
 	const match = inner.router.getMatch(matchId);
+	if (!match) return inner.matches[index];
 	if (!loaderIsRunningAsync) {
 		match._nonReactive.loaderPromise?.resolve();
 		match._nonReactive.loadPromise?.resolve();
@@ -500,7 +503,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 			isFetching: nextIsFetching,
 			invalid: false
 		}));
-		return inner.router.getMatch(matchId);
+		return inner.router.getMatch(matchId) ?? inner.matches[index];
 	} else return match;
 };
 async function loadMatches(arg) {
diff --git a/dist/esm/load-matches.js b/dist/esm/load-matches.js
index bcea2e0..fd9ebbf 100644
--- a/dist/esm/load-matches.js
+++ b/dist/esm/load-matches.js
@@ -439,6 +439,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 				try {
 					await runLoader(inner, matchPromises, matchId, index, route);
 					const match = inner.router.getMatch(matchId);
+					if (!match) return;
 					match._nonReactive.loaderPromise?.resolve();
 					match._nonReactive.loadPromise?.resolve();
 					match._nonReactive.loaderPromise = void 0;
@@ -469,6 +470,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 			if (prevMatch.status === "success" && !inner.sync && !prevMatch.preload && shouldReloadInBackground) return prevMatch;
 			await prevMatch._nonReactive.loaderPromise;
 			const match = inner.router.getMatch(matchId);
+			if (!match) return inner.matches[index];
 			const error = match._nonReactive.error || match.error;
 			if (error) handleRedirectAndNotFound(inner, match, error);
 			if (match.status === "pending") await handleLoader(preload, prevMatch, previousRouteMatchId, match, route);
@@ -484,6 +486,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 		}
 	}
 	const match = inner.router.getMatch(matchId);
+	if (!match) return inner.matches[index];
 	if (!loaderIsRunningAsync) {
 		match._nonReactive.loaderPromise?.resolve();
 		match._nonReactive.loadPromise?.resolve();
@@ -500,7 +503,7 @@ const loadRouteMatch = async (inner, matchPromises, index) => {
 			isFetching: nextIsFetching,
 			invalid: false
 		}));
-		return inner.router.getMatch(matchId);
+		return inner.router.getMatch(matchId) ?? inner.matches[index];
 	} else return match;
 };
 async function loadMatches(arg) {
diff --git a/src/load-matches.ts b/src/load-matches.ts
index f901a0c..76674e0 100644
--- a/src/load-matches.ts
+++ b/src/load-matches.ts
@@ -835,7 +835,10 @@ const loadRouteMatch = async (
       ;(async () => {
         try {
           await runLoader(inner, matchPromises, matchId, index, route)
-          const match = inner.router.getMatch(matchId)!
+          const match = inner.router.getMatch(matchId)
+          if (!match) {
+            return
+          }
           match._nonReactive.loaderPromise?.resolve()
           match._nonReactive.loadPromise?.resolve()
           match._nonReactive.loaderPromise = undefined
@@ -903,7 +906,10 @@ const loadRouteMatch = async (
         return prevMatch
       }
       await prevMatch._nonReactive.loaderPromise
-      const match = inner.router.getMatch(matchId)!
+      const match = inner.router.getMatch(matchId)
+      if (!match) {
+        return inner.matches[index]!
+      }
       const error = match._nonReactive.error || match.error
       if (error) {
         handleRedirectAndNotFound(inner, match, error)
@@ -933,7 +939,10 @@ const loadRouteMatch = async (
       await handleLoader(preload, prevMatch, previousRouteMatchId, match, route)
     }
   }
-  const match = inner.router.getMatch(matchId)!
+  const match = inner.router.getMatch(matchId)
+  if (!match) {
+    return inner.matches[index]!
+  }
   if (!loaderIsRunningAsync) {
     match._nonReactive.loaderPromise?.resolve()
     match._nonReactive.loadPromise?.resolve()
@@ -952,7 +961,7 @@ const loadRouteMatch = async (
       isFetching: nextIsFetching,
       invalid: false,
     }))
-    return inner.router.getMatch(matchId)!
+    return inner.router.getMatch(matchId) ?? inner.matches[index]!
   } else {
     return match
   }
