diff --git a/unpack.js b/unpack.js
index 2e8fc87..75305e1 100644
--- a/unpack.js
+++ b/unpack.js
@@ -491,25 +491,8 @@ export function read() {
 		}
 	}
 }
-const validName = /^[a-zA-Z_$][a-zA-Z\d_$]*$/;
 function createStructureReader(structure, firstId) {
 	function readObject() {
-		// This initial function is quick to instantiate, but runs slower. After several iterations pay the cost to build the faster function
-		if (readObject.count++ > inlineObjectReadThreshold) {
-			let optimizedReadObject;
-			try {
-				optimizedReadObject = structure.read = (new Function('r', 'return function(){return ' + (currentUnpackr.freezeData ? 'Object.freeze' : '') +
-					'({' + structure.map(key => key === '__proto__' ? '__proto_:r()' : validName.test(key) ? key + ':r()' : ('[' + JSON.stringify(key) + ']:r()')).join(',') + '})}'))(read);
-			} catch(error) {
-				// in CF workers, the new Function call could begin to fail at any point in time
-				inlineObjectReadThreshold = Infinity; // disable going forward
-				return readObject(); // recursively try again
-			}
-			structure.read0 = optimizedReadObject; // keep the un-wrapped body reader in sync
-			if (structure.highByte === 0)
-				structure.read = createSecondByteReader(firstId, structure.read);
-			return optimizedReadObject(); // second byte is already read, if there is one so immediately read object
-		}
 		let object = {};
 		for (let i = 0, l = structure.length; i < l; i++) {
 			let key = structure[i];
